### PR TITLE
NIAD-280: encounter appointment

### PR DESCRIPTION
**Pull request by Alicja-Kainos**

### CODE REVIEW

File: service/src/integration-test/resources/fhir/encounter_valid.json [hunk 1]
```diff
@@ -87,6 +90,46 @@
...
+        "minutesDuration": 10,
+        "comment": "\nDate/Time: 01 June 2017 14:00\nLocation: CENTRAL LONDON COMMUNITY HEALTHCARE NHS TRUST\nPsychiatric Community Care Service\n                                            ",
+        "participant": [
+          {
+            "required": "required",
+            "status": "accepted"
+          },
+          {
+            "actor": {
+              "reference": "#1"
```
Comment (dawidskainos) [inline service/src/integration-test/resources/fhir/encounter_valid.json:126]: It does not look like a valid reference

File: service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/EncounterMapper.java [hunk 2]
```diff
@@ -68,4 +74,15 @@ private Period getPeriod(POCDMT000002UK01ClinicalDocument1 clinicalDocument) {
...
     }
+
+    private void getAppointment(Encounter encounter, POCDMT000002UK01ClinicalDocument1 clinicalDocument) {
+        Reference referralRequest = encounter.getIncomingReferralFirstRep();
+        Reference patient = encounter.getSubject();
+
+        Optional<Appointment> appointment = appointmentService.retrieveAppointment(referralRequest, patient, clinicalDocument);
+        if (appointment.isPresent()) {
+            encounter.setAppointment(new Reference(appointment.get()));
+            encounter.setAppointmentTarget(appointment.get());
```
Comment (dawidskainos) [inline service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/EncounterMapper.java:85]: Isn't it duplicated with EncounterReportBundleService?

File: service/src/integration-test/resources/fhir/encounter_valid.json (hunk 1 above)
Comment (dawidskainos) [inline service/src/integration-test/resources/fhir/encounter_valid.json:126]: It does not look like a valid reference

File: service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/service/EncounterReportBundleService.java [hunk 3]
```diff
@@ -52,12 +57,23 @@ private void addServiceProvider(Bundle bundle, Encounter encounter) {
...
     }
+
+    private void addAppointment(Bundle bundle, Encounter encounter) {
+        Appointment appointment = encounter.getAppointmentTarget();
+        if (appointment != null) {
+            bundle.addEntry()
+                .setFullUrl(appointment.getIdElement().getValue())
+                .setResource(appointment);
+            encounter.setAppointment(new Reference(appointment));
+            encounter.setAppointmentTarget(appointment);
```
Comment (dawidskainos) [inline service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/service/EncounterReportBundleService.java:76]: What is the purpose for setting appointmentTarget?

File: service/src/integration-test/resources/fhir/encounter_valid.json (hunk 1 above)
Comment (Alicja-Kainos) [reply service/src/integration-test/resources/fhir/encounter_valid.json:126]: your're right :) heh XD

File: service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/EncounterMapper.java (hunk 2 above)
Comment (Alicja-Kainos) [reply service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/EncounterMapper.java:85]: good point

File: service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/service/EncounterReportBundleService.java (hunk 3 above)
Comment (Alicja-Kainos) [reply service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/service/EncounterReportBundleService.java:76]: there should not be the setting of Appointment at all - I did not spot it after merging :) The appointment itself is taken from encounter :)

File: service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/EncounterMapper.java (hunk 2 above)
Comment (Alicja-Kainos) [reply service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/EncounterMapper.java:85]: it should be only here int the mapper

File: service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/EncounterMapper.java [hunk 4]
```diff
@@ -30,12 +33,15 @@
...
 
+    private AppointmentService appointmentService;
+
     public Encounter mapEncounter(POCDMT000002UK01ClinicalDocument1 clinicalDocument) {
         Encounter encounter = new Encounter();
         encounter.setIdElement(newRandomUuid());
         encounter.setStatus(FINISHED);
         encounter.setParticipant(getEncounterParticipantComponents(clinicalDocument));
         encounter.setPeriod(getPeriod(clinicalDocument));
+        getAppointment(encounter, clinicalDocument);
```
Comment (dawidskainos) [inline service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/EncounterMapper.java:44]: I think "setAppointment" would make more sense

File: service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/service/EncounterReportBundleService.java [hunk 5]
```diff
@@ -23,6 +25,8 @@
 
     private ServiceProviderMapper serviceProviderMapper;
 
+    private AppointmentService appointmentService;
```
Comment (dawidskainos) [inline service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/service/EncounterReportBundleService.java:28]: Not necessary field.

**Pull request merged**